compat/wserrno.cc

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 1996-2025 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/wserrno.h"
+
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_
+
+#include <unordered_map>
+
+void
+SetErrnoFromWsaError()
+{
+    // POSIX codes which socket API users may care about
+    static const auto *CodeMap = new std::unordered_map<int, int> {
+
+        { WSAECONNABORTED, ECONNABORTED },
+
+        { WSAEINPROGRESS, EINPROGRESS },
+        { WSAEAFNOSUPPORT, EAFNOSUPPORT },
+        { WSAEINVAL, EINVAL },
+        { WSAEISCONN, EISCONN },
+
+        { WSAEWOULDBLOCK, EWOULDBLOCK },
+        // Windows does not have a equivalent of EAGAIN
+        { WSAEALREADY, EALREADY },
+        { WSAEINTR, EINTR },
+        // no Windows error code maps to ERESTART
+
+        { WSAEMFILE, EMFILE },
+        // no Windows error code maps to ENFILE
+
+        { WSAECONNRESET, ECONNRESET }
+    };
```

Comment:
I recommend sorting `{WSAX, EY}` entries alphabetically so that it is easier to find a particular error and easier to decide where to place new entries. I would sort by the second/EY field (so that comments can participate in the same sort rather than moved to the beginning or end), but sorting by the first/WSAX field is better than no sorting at all.

I do not insist on this, but the current grouping looks odd. I am aware of this grouping origins.

Reply:
The list is short enough that sorting shouldn't be a problem

Reply:
I disagree that humans are good at (looking for and) finding matches among a dozen of similar-looking values (not all of which are going to be exposed in a diff that reviewers will see!), but assisting humans with search and duplicate avoidance is not what _this_ change request is focused on: The primary problem flagged by this change request is not lack of sorting but odd-looking grouping.

Sorting is just a possible way to address that primary problem. IMHO, sorting is good because it has positive side effects that go beyond removing odd groups, but I do not insist on that particular solution.

Reply:
Sorry, I had not understood the request. Grouping removed.